File: VaquinhaAnimal.Api/VaquinhaAnimal.Domain/Entities/Campanha.cs
using VaquinhaAnimal.Domain.Entities.Base;
using VaquinhaAnimal.Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace VaquinhaAnimal.Domain.Entities
{
    public class Campanha : BaseEntity
    {
        [JsonPropertyName("data_criacao")]
        [Required]
        public DateTime DataCriacao { get; set; }

        [JsonPropertyName("tipo_campanha")]
        [Required]
        public TipoCampanhaEnum TipoCampanha { get; set; }

        [JsonPropertyName("data_inicio")]
        public DateTime? DataInicio { get; set; }

        [JsonPropertyName("tag_campanha")]
        public TagCampanhaEnum TagCampanha { get; set; }

        [JsonPropertyName("duracao_dias")]
        public int? DuracaoDias { get; set; } // SE FOR RECORRENTE NÃO TEM DURAÇÃo

        [JsonPropertyName("data_encerramento")]
        public DateTime? DataEncerramento { get; set; }

        [JsonPropertyName("titulo")]
        [Required]
        public string Titulo { get; set; }

        [JsonPropertyName("video_url")]
        public string VideoUrl { get; set; }

        [JsonPropertyName("descricao_curta")]
        [Required]
        public string DescricaoCurta { get; set; } // (200)

        [JsonPropertyName("url_campanha")]
        [Required]
        public string UrlCampanha { get; set; } // (200)

        [JsonPropertyName("descricao_longa")]
        [Required]
        public string DescricaoLonga { get; set; }// (MÍNIMO 500 MÁXIMO 5.000)

        [JsonPropertyName("valor_desejado")]
        [Required]
        public decimal ValorDesejado { get; set; } // MAIOR QUE ZERO

        [JsonPropertyName("total_arrecadado")]
        [Required]
        public decimal TotalArrecadado { get; set; } // MAIOR OU IGUAL A ZERO

        [JsonPropertyName("termos")]
        [Required]
        public bool Termos { get; set; }

        [JsonPropertyName("premium")]
        [Required]
        public bool Premium { get; set; }

        [JsonPropertyName("status_campanha")]
        [Required]
        public StatusCampanhaEnum StatusCampanha { get; set; }

        [JsonPropertyName("usuario_id")]
        [Required]
        public Guid Usuario_Id { get; set; }

        [JsonPropertyName("imagens")]
        public List<Imagem> Imagens { get; set; }

        [JsonPropertyName("beneficiario")]
        public Beneficiario Beneficiario { get; set; }

        [JsonIgnore]
        [JsonPropertyName("doacoes")]
        public List<Doacao> Doacoes { get; set; }

        [JsonIgnore]
        [JsonPropertyName("assinaturas")]
        public List<Assinatura> Assinaturas { get; set; }
    }
}

File: VaquinhaAnimal.App/src/app/campanhas/model/Campanha.ts
export class Campanha {
    id!: string;
    data_inicio?: Date;
    data_encerramento?: Date;
    duracao_dias?: number;
    tag_campanha!: TagCampanhaEnum;
    tipo_campanha!: TipoCampanhaEnum;
    titulo!: string;    
    descricao_curta!: string;
    descricao_longa!: string;
    url_campanha!: string;
    valor_desejado!: number;
    total_arrecadado?: number;
    termos!: boolean;
    premium!: boolean;
    status_campanha?: number;
    percentual_arrecadado?: number;
    restam?: number;
    totalDoadores?: number;
    usuario_id!: string;
    video_url!: string;
    
    beneficiario!: Beneficiario;
    imagens!: Imagem[];
}

export class Beneficiario {
    id!: string;
    nome!: string;
    documento!: string;
    tipo!: string;
    codigo_banco!: string;
    numero_agencia!: string;
    digito_agencia!: string;
    numero_conta!: string;
    digito_conta!: string;
    tipo_conta!: string;
    recebedor_id!: string;
    campanha_id!: string;
}

export class Imagem {
    id!: string;
    tipo!: TipoImagemEnum;
    arquivo!: string;
    arquivo_upload!: string;
    campanha_id!: string;
}

export enum TipoImagemEnum {
    Capa = 1,
    Interna = 2
}

export enum TagCampanhaEnum {
    Alimentacao = 1,
    Medicamentos = 2,
    Cirurgias = 3,
    Tratamentos = 4,
    Acessórios = 5,
    PuroAmor = 6,
    Outros = 7
}

export enum TipoCampanhaEnum {
    Recorrente = 1,
    Unica = 2
}